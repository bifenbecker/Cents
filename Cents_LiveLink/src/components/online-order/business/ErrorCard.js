import React from "react";
import {Flex, Text, Image, Card} from "rebass/styled-components";
import {ClothesPile} from "../../../assets/images/index";

const ErrorCard = () => {
  return (
    <Card {...styles.cardContainer}>
      <Flex {...styles.contentContainer}>
        <Flex {...styles.textContainer}>
          <Text {...styles.text.headerText}>
            Sorry, we can't pick up from this address
          </Text>
        </Flex>
        <Flex {...styles.imageContainer}>
          <Image src={ClothesPile} {...styles.imageStyles}></Image>
        </Flex>
      </Flex>

      <Flex {...styles.footer.wrapper}>
        <Text {...styles.footer.text}>
          Your address is not in our delivery service area. If you'd like to drop off your
          laundry at our laundromat, we'll be happy to serve you!
        </Text>
      </Flex>
    </Card>
  );
};

const styles = {
  cardContainer: {
    sx: {
      background: "rgba(220, 86, 46, 0.1)",
      width: "calc(100% - 40px)",
      minHeight: "170px",
      p: "18px",
      border: "1.5px solid #DC562E",
      borderRadius: "12px",
      boxShadow:
        "0 4px 5px 0 rgba(0,0,0,0.14), 0 1px 10px 0 rgba(0,0,0,0.12), 0 2px 4px -1px rgba(0,0,0,0.2)",
      my: "18px",
      mx: "20px",
    },
  },
  textContainer: {
    flex: "1",
    fontSize: "18px",
  },
  text: {
    headerText: {
      fontSize: "18px",
      fontWeight: "700",
      color: "#303651",
    },
  },
  imageStyles: {
    height: "80%",
  },
  imageContainer: {
    flex: "1",
    justifyContent: "center",
    alignItems: "center",
  },
  contentContainer: {
    height: "92px",
    justifyContent: "space-between",
  },
  footer: {
    wrapper: {
      width: "100%",
      mt: "auto",
      alignItems: "flex-end",
      justifyContent: "center",
    },
    text: {
      fontFamily: "secondary",
      color: "#303651",
      fontWeight: "400",
      fontSize: "14px",
    },
  },
};

export default ErrorCard;
