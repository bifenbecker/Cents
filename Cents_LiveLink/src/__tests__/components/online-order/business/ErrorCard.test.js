import React from "react";
import {render} from "@testing-library/react";
import ErrorCard from "../../../../components/online-order/business/ErrorCard";

describe("Testing ErrorCard Component", () => {
  it("Test errorCard UI snapshot", async () => {
    expect(render(<ErrorCard />)).toMatchSnapshot();
  });
});
