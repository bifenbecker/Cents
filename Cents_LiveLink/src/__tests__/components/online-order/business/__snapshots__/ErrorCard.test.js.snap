// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ErrorCard Component Test errorCard UI snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bczRLJ hhoFwm"
      >
        <div
          class="sc-bczRLJ sc-gsnTZi bNSgyd hzPtvw"
          height="92px"
        >
          <div
            class="sc-bczRLJ sc-gsnTZi hsphLw hzPtvw"
            font-size="18px"
          >
            <div
              class="sc-bczRLJ eQobRH"
              font-size="18px"
            >
              Sorry, we can't pick-up from this address :(
            </div>
          </div>
          <div
            class="sc-bczRLJ sc-gsnTZi hdFran hzPtvw"
          >
            <img
              class="sc-bczRLJ kHAieo"
              height="80%"
              src="Pile_Of_Clothes.svg"
            />
          </div>
        </div>
        <div
          class="sc-bczRLJ sc-gsnTZi evAOmX hzPtvw"
          width="100%"
        >
          <div
            class="sc-bczRLJ fjmcRP"
            font-family="secondary"
          >
            Your address is not in our delivery service area. If you'd like to drop off your laundry at our laundromat, we'll be happy to serve you!
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bczRLJ hhoFwm"
    >
      <div
        class="sc-bczRLJ sc-gsnTZi bNSgyd hzPtvw"
        height="92px"
      >
        <div
          class="sc-bczRLJ sc-gsnTZi hsphLw hzPtvw"
          font-size="18px"
        >
          <div
            class="sc-bczRLJ eQobRH"
            font-size="18px"
          >
            Sorry, we can't pick-up from this address :(
          </div>
        </div>
        <div
          class="sc-bczRLJ sc-gsnTZi hdFran hzPtvw"
        >
          <img
            class="sc-bczRLJ kHAieo"
            height="80%"
            src="Pile_Of_Clothes.svg"
          />
        </div>
      </div>
      <div
        class="sc-bczRLJ sc-gsnTZi evAOmX hzPtvw"
        width="100%"
      >
        <div
          class="sc-bczRLJ fjmcRP"
          font-family="secondary"
        >
          Your address is not in our delivery service area. If you'd like to drop off your laundry at our laundromat, we'll be happy to serve you!
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
